src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -418,6 +416,49 @@ class WorkspaceService(dataSource: DataSource, workspaceDAO: WorkspaceDAO, entit
       // to ensure it is evaluated within the transaction
       RequestComplete(methodConfigurationDAO.list(workspaceNamespace, workspaceName, txn).toList)
     }
+
+  // TODO: replace this stub with the actual expression parser
+  private def resolveInputs( methodConfig: MethodConfiguration, entity: Entity ): String =
+    """{"hello": "world"}"""
+
+  def submitJob(workspaceName: WorkspaceName, jobDesc: JobDescription, authCookie: HttpCookie): PerRequestMessage =
+    dataSource inTransaction { txn =>
+      methodConfigurationDAO.get(workspaceName.namespace, workspaceName.name, jobDesc.methodConfigurationNamespace, jobDesc.methodConfigurationName, txn) match {
+        case None =>
+          RequestComplete(StatusCodes.NotFound,
+            s"No method configuration named ${jobDesc.methodConfigurationNamespace}/${jobDesc.methodConfigurationName} in workspace ${workspaceName.namespace}/${workspaceName.name}")
+        case Some(methodConfig) =>
+          if ( methodConfig.rootEntityType != jobDesc.entityType )
+            RequestComplete(StatusCodes.Conflict,
+                s"The method configuration expects an entity of type ${methodConfig.rootEntityType}, but the request describes an entity of type ${jobDesc.entityType}")
+          else {
+            entityDAO.get(workspaceName.namespace, workspaceName.name, jobDesc.entityType, jobDesc.entityName, txn) match {
+              case None =>
+                RequestComplete(StatusCodes.NotFound,
+                  s"No entity of type ${jobDesc.entityType} named ${jobDesc.entityName} in workspace in workspace ${workspaceName.namespace}/${workspaceName.name}")
+              case Some(entity) =>
+                Try { resolveInputs(methodConfig,entity) } match {
+                  case Failure(error) =>
+                    RequestComplete(StatusCodes.Conflict, "Expression evaluation failed: "+error.getMessage())
+                  case Success(inputs) =>
+                    methodRepoDAO.getMethod(methodConfig.methodNamespace, methodConfig.methodName, methodConfig.methodVersion, authCookie) match {
+                      case None => RequestComplete(StatusCodes.NotFound, "Can't get method from Method Repo")
+                      case Some(agoraEntity) =>
+                        if ( agoraEntity.payload.isEmpty )
```

Comment:
I started out matching Success and Failure where it was more terse to use pattern matching to extract the error or the object.  They could all be switched to "if" at this point.  But since it's less typing, I'll change the "if" to a match.